src/main/archetype/ui.tests/test-module/specs/aem/forms.js

```diff
@@ -0,0 +1,153 @@
+            ],
+            getContainerSelector = (templateContentPath, container) => {
+                return `div[data-path="${templateContentPath}/${container}"]`;
+            },
+            verifyTemplate = (templatePath, isBlankForm) => {
+                let templateStructure = templatePath + '/structure/jcr:content',
+                    templateInitialContent = templatePath + '/initial/jcr:content',
+                    afContainerOverlaySelector = '#OverlayWrapper ' + getContainerSelector(templateInitialContent,'guideContainer');
+
+                //Open template in editing mode
+                browser.url(`${config.aem.author.base_url}/editor.html${templatePath}/structure.html`);
+
+                //First open structure layer
+                expect($(selectors.editor.layerSwitcher).waitForDisplayed()).toBe(true);
+                $(selectors.editor.layerSwitcher).click();
+                expect($(selectors.editor.layerSelector.structure).waitForDisplayed()).toBe(true);
+                $(selectors.editor.layerSelector.structure).click();
+
+                //Verify all three containers are visible
+                expect($(getContainerSelector(templateStructure,'parsys1'))).toBeDisplayed();
+                expect($(getContainerSelector(templateStructure,'parsys2'))).toBeDisplayed();
+                expect($(getContainerSelector(templateStructure,'guideContainer'))).toBeDisplayed();
+
+                //Verify Applied Policy Content
+                $(getContainerSelector(templateStructure,'guideContainer')).click();
+                expect($(selectors.editor.templateEditor.structureLayer.policyButton).waitForDisplayed()).toBe(true);
+                $(selectors.editor.templateEditor.structureLayer.policyButton).click();
+                expect($(selectors.editor.templateEditor.structureLayer.policyPage.allowedComponent.adaptiveFormGroup + '[checked]').waitForDisplayed()).toBe(true);
+                expect($(selectors.editor.templateEditor.structureLayer.policyPage.cancel).waitForDisplayed()).toBe(true);
+                $(selectors.editor.templateEditor.structureLayer.policyPage.cancel).click();
+
+                //Switch to initial content layer
+                expect($(selectors.editor.layerSwitcher).waitForDisplayed()).toBe(true);
+                $(selectors.editor.layerSwitcher).click();
+                expect($(selectors.editor.layerSelector.initial).waitForDisplayed()).toBe(true);
+                $(selectors.editor.layerSelector.initial).click();
+                expect($(afContainerOverlaySelector).waitForDisplayed()).toBe(true);
+
+                //Verify no console error present in structure and initial content layer
+                expect(consoleErrors).toHaveLength(0);
+
+                //Open Side panel, verify that no invalid item in properties tab and then close the side panel
+                expect($(selectors.editor.sidePanelToggleButton).waitForDisplayed()).toBe(true);
```

Comment:
Let's keep the test simple and only use documented / public constructs, we don't need to open side panel or properties (these are already part of produce test suite). This test should ideally ensure the following,
1. Content is editor compatible - just verify if the content is usable in editor and check for any console errors
2. Preview of content (ie) actual business logic works as expected

Reply:
@rismehta, We are following the below mentioned strategy for content validation.

**Template**

1. Open in Editing Mode, verify console error
2. Structure Layer- All three containers (2 parsys, guidecontainer) presence
3. Structure Layer - Applied Policy verification
4. Switch to Initial Content Layer, verify console error
5. Preview/Runtime - Tabbed Layout, Navigating through tabs and verifying no. of AF field presence based on a rule file. Rule files is showing panel and fields in the hierarchical structure. Panel can contain multiple child panels (JSONArray to contain multiple panel), or AF fields (JSONObject with count property for total no. of fields)

**Theme**

1. Open Theme in editing mode, verify console error
2. Preview/Runtime - verify actually applied css rules to AF components against a css rule file. These rules contains only those styling which applied through theme only. These rules can be different for different themes.

**Console Error**
Console errors are verified using puppeteer as there is no API support for reading console error in WebdriverIO v7. As per https://webdriver.io/docs/automationProtocols WebdriverIO supports Chrome DevTools Protocol using DevTools NPM package.

**Note**
These added tests are not being executed as part of PR execution. In PR execution, only integration tests are executed - https://github.com/adobe/aem-project-archetype/blob/develop/.github/workflows/maven-test.yml#L58
Once this PR is merged, these tests will be executed as part of pipeline execution for Deploy to CM github action - https://github.com/adobe/aem-project-archetype/blob/develop/.github/workflows/maven-deploy-cm.yml#L33

Reply:
Some editor steps removed as console error verification has been removed which was the sole objective of the rendering editor ui, updated the status in the above comments. Now only content preview/runtime verification.
cc : @rismehta , @catalan-adobe